src/openforms/formio/registry.py

```diff
@@ -27,22 +27,21 @@
 if TYPE_CHECKING:
     from openforms.submissions.models import Submission
 
+T = TypeVar("T", bound=Component, contravariant=True)
 
-class FormatterProtocol(Protocol):
-    def __init__(self, as_html: bool): ...
-
-    def __call__(self, component: Component, value: Any) -> str: ...
 
+class FormatterProtocol(Generic[T], Protocol):
```

Comment:
You can use the shorthand `Protocol[T]` here, this is a special case for protocols

Reply:
TIL, didn't know that was already in 3.10

Reply:
Haven't thought about it until now, but it does look similar to the new 3.12 syntax